### PR TITLE
fix(dav): Reduce CalDAV backend memory footprint

### PR DESCRIPTION
**Pull request by ChristophWurst**
## Summary

fetchAll inflates memory. Fetching in a loop allows GC to run earlier and more often.

## TODO

- [x] Replace memory-hungry fetchAll

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by miaulalala — APPROVED** [review]

File: apps/dav/lib/CalDAV/CalDavBackend.php [hunk 1]
```diff
@@ -2656,9 +2661,9 @@ public function getSchedulingObjects($principalUri) {
 				->where($query->expr()->eq('principaluri', $query->createNamedParameter($principalUri)))
 				->executeQuery();
 
-		$result = [];
-		foreach ($stmt->fetchAll() as $row) {
```
Comment (ChristophWurst) [inline apps/dav/lib/CalDAV/CalDavBackend.php:2660]: from prod:

```json
{
  "reqId": "abc123",
  "level": 3,
  "time": "2023-10-06T06:25:16+00:00",
  "remoteAddr": "aaa:bbb:ccc::",
  "user": "user123",
  "app": "PHP",
  "method": "REPORT",
  "url": "/remote.php/dav/calendars/user123/inbox/",
  "message": "Allowed memory size of 536870912 bytes exhausted (tried to allocate 135168 bytes) at /var/www/nextcloud/apps/dav/lib/CalDAV/CalDavBackend.php#2661",
  "userAgent": "iOS/17.0.3 (21A360) dataaccessd/1.0",
  "version": "27.1.2.1",
  "data": {
    "app": "PHP"
  },
  "id": "651faa6e0d5c0"
}
```

Comment (ChristophWurst) [reply apps/dav/lib/CalDAV/CalDavBackend.php:2660]: yet this probably means memory was almost full before. those 135168 are just the tip of the iceberg.

Comment (blizzz) [reply apps/dav/lib/CalDAV/CalDavBackend.php:2660]: all the array_(map|filter|…) functions are also quite memory heave

**Review by blizzz — APPROVED** [review]

**Review by kesselb — APPROVED** [review]
:racing_car:

**Comment by ChristophWurst** [comment]
It's there.

**Pull request merged**